### ISSUE TITLE
bug: ion-segment a11y - role should be tablist, buttons should use aria-selected

### ISSUE THREAD

**Issue by eric-horodyski** [ionic-team/ionic-framework#23059]
# Bug Report

**Ionic version:**

[ ] **4.x**
[x] **5.x**

**Current behavior:**

When using `ion-segment` there is no accessibility role defined to the component. Inner `ion-segment-button` components set `aria-pressed` to either true or false, depending on their selected status.

**Expected behavior:**

It's expected that `ion-segment` should have `role="tablist"` and each `ion-segment-button` should have `role="tab"` while setting `aria-selected` to either true or false, depending on their selected status.

Additionally, it would be nice if there was the ability to set `aria-controls` on an `ion-segment-button`. 

**Steps to reproduce:**

1. Pull repository below
2. Run `npm install && npm run start`
3. Inspect `ion-segment` and `ion-segment-button` components

**Related code:**

https://github.com/eric-horodyski/fw-segment-a11y-issue

**Ionic info:**

```
Ionic:

   Ionic CLI       : 6.13.1 (/Users/eric/npm-global/lib/node_modules/@ionic/cli)
   Ionic Framework : @ionic/react 5.6.0

Capacitor:

   Capacitor CLI   : 2.4.7
   @capacitor/core : 2.4.7

Utility:

   cordova-res (update available: 0.15.3) : 0.15.2
   native-run (update available: 1.3.0)   : 1.2.2

System:

   NodeJS : v14.15.1 (/usr/local/bin/node)
   npm    : 6.14.11
   OS     : macOS Big Sur
```
